godot-core/src/builtin/color_hsv.rs

```diff
@@ -0,0 +1,296 @@
+        (wrap_hue(self.h - other.h).is_zero_approx())
+            && (self.s - other.s).abs().is_zero_approx()
+            && (self.v - other.v).abs().is_zero_approx()
+            && (self.a - other.a).abs().is_zero_approx()
+    }
+}
+
+impl ColorHsv {
+    /// Construct from Hue, Saturation and Value.
+    ///
+    /// Alpha will be set at `1.` by default. To construct with custom Alpha value, use [`ColorHsv::from_hsva`] constructor.
+    pub const fn from_hsv(h: f32, s: f32, v: f32) -> Self {
+        ColorHsv { h, s, v, a: 1.0 }
+    }
+
+    /// Construct from Hue, Saturation, Value and Alpha.
+    ///
+    /// To construct with Alpha set as default `1.`, use [`ColorHsv::from_hsv`] constructor.
+    pub const fn from_hsva(h: f32, s: f32, v: f32, a: f32) -> Self {
+        ColorHsv { h, s, v, a }
+    }
+
+    /// Transforms the `ColorHsv` into one with values clamped to the range valid for transformation into [`Color`].
+    ///
+    /// To normalize with **Hue** value wrapped, not clamped (for continuity around the hue wheel), use [`ColorHsv::normalized_wrapped_h`].
+    ///
+    /// ## Example
+    ///
+    /// ```
+    /// use godot::builtin::ColorHsv;
+    /// use godot::builtin::math::assert_eq_approx;
+    ///
+    /// let hsv_c = ColorHsv::from_hsv(1.35, -0.60, 1.15);
+    /// let normalized = hsv_c.normalized_clamped_h();
+    /// assert_eq_approx!(normalized, ColorHsv::from_hsv(1.0, 0.0, 1.0));
+    /// ```
+    #[must_use]
+    pub fn normalized_clamped_h(self) -> Self {
+        ColorHsv {
+            h: self.h.clamp(0., 1.),
+            s: self.s.clamp(0., 1.),
+            v: self.v.clamp(0., 1.),
+            a: self.a.clamp(0., 1.),
+        }
+    }
+
+    /// Transforms the `ColorHsv` into one with **Hue** value wrapped and SVA clamped to the range valid for transformation into [`Color`].
+    ///
+    /// To normalize with **Hue** value clamped in the same way as SVA, use [`ColorHsv::normalized_clamped_h`].
+    ///
+    /// ## Example
+    ///
+    /// ```
+    /// use godot::builtin::ColorHsv;
+    /// use godot::builtin::math::assert_eq_approx;
+    ///
+    /// let hsv_c = ColorHsv::from_hsv(1.35, -0.60, 1.15);
+    /// let normalized = hsv_c.normalized_wrapped_h();
+    /// assert_eq_approx!(normalized, ColorHsv::from_hsv(0.35, 0.0, 1.0));
+    /// ```
+    #[must_use]
+    pub fn normalized_wrapped_h(self) -> Self {
+        ColorHsv {
+            h: wrap_hue(self.h),
+            s: self.s.clamp(0., 1.),
+            v: self.v.clamp(0., 1.),
+            a: self.a.clamp(0., 1.),
+        }
+    }
+
+    /// Convert `ColorHsv` into [`Color`].
+    ///
+    /// ## Panics
+    ///
+    /// Method will panic if the HSVA values are outside of the valid range `0.0..=1.0`. You can use [`ColorHsv::normalized_clamped_h`] or
+    /// [`ColorHsv::normalized_wrapped_h`] to ensure they are in range, or use [`ColorHsv::try_to_rgb`] implementation.
+    #[must_use]
+    pub fn to_rgb(self) -> Color {
```

Comment:
We can add the warning sign here (see explanation [here](https://godot-rust.github.io/docs/gdext/master/godot/#ergonomics-and-panics)). `Panics` should be H1.
```suggestion
    /// ⚠️ Convert `ColorHsv` into [`Color`].
    ///
    /// # Panics
    ///
    /// Method will panic if the HSVA values are outside of the valid range `0.0..=1.0`. You can use [`ColorHsv::normalized_clamped_h`] or
    /// [`ColorHsv::normalized_wrapped_h`] to ensure they are in range, or use [`ColorHsv::try_to_rgb`] implementation.
    pub fn to_rgb(self) -> Color {
```